flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -108,7 +110,7 @@ public RemoteInputChannel(
 		ResultPartitionID partitionId,
 		ConnectionID connectionId,
 		ConnectionManager connectionManager,
-		TaskIOMetricGroup metrics) {
+		ProxyMetricGroup metrics) {
```

Comment:
I would suggest creating `TaskNetworkMetrics` internally in for `NetworkEnviroment` (now in `registerTask`) and create/register all metrics there, similar to how it is done in `TaskIOMetricGroup`. Then pass only what is needed to gate and channels. This way we could keep metrics concern separately. All aggregating gauges from `TaskIOMetricGroup.initializeBufferMetrics` could be also moved from `TaskIOMetricGroup` into `TaskNetworkMetrics` (separate commit or PR for FLINK-12202).